Modify the definition of long type in swigfaiss.swig to avoid build errors in swig 4.2.
The following MR has been merged into the main branch of faiss, so this patch will not be needed in the next version of faiss v1.8.0.
Ref. https://github.com/facebookresearch/faiss/pull/3315

Copyright (c) 2024 Di-Is

This software is released under the MIT License.
http://opensource.org/licenses/mit-license.php

diff --git a/faiss/faiss/python/swigfaiss.swig b/faiss/faiss/python/swigfaiss.swig
index fb7f50dd..0ea93609 100644
--- a/faiss/faiss/python/swigfaiss.swig
+++ b/faiss/faiss/python/swigfaiss.swig
@@ -1022,14 +1022,14 @@ PyObject *swig_ptr (PyObject *a)
         return SWIG_NewPointerObj(data, SWIGTYPE_p_bool, 0);
     }
     if(PyArray_TYPE(ao) == NPY_UINT64) {
-#ifdef SWIGWORDSIZE64
+#if (__SIZEOF_LONG__ == 8)
         return SWIG_NewPointerObj(data, SWIGTYPE_p_unsigned_long, 0);
 #else
         return SWIG_NewPointerObj(data, SWIGTYPE_p_unsigned_long_long, 0);
 #endif
     }
     if(PyArray_TYPE(ao) == NPY_INT64) {
-#ifdef SWIGWORDSIZE64
+#if  (__SIZEOF_LONG__ == 8)
         return SWIG_NewPointerObj(data, SWIGTYPE_p_long, 0);
 #else
         return SWIG_NewPointerObj(data, SWIGTYPE_p_long_long, 0);
